src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -456,6 +457,14 @@ private function resolveVariables($value)
                 $value = (string) getenv($name);
             }
 
+            if ('' === $value && isset($matches['default_value'])) {
+                if (strpbrk($matches['default_value'], '\'"{}$')) {
+                    throw $this->createFormatException('Invalid character in default value');
```

Comment:
`sprintf('Unsupported character found in the default value of variable "$%s".', $name))`

Reply:
Aah yes, much better 👍